File: Production-SPA/src/app/models/plan-item.ts
import { Product } from '../models/product';

export class PlanItem {
  id: number;
  annualProductionPlanId: number;
  quantity: number;
  description: string;
  productId: number;
  product: Product;

  constructor(p?: any) {
    if (p == null) {
      return;
    }
    this.id = p.id;
    this.annualProductionPlanId = p.annualProductionPlanId;
    this.quantity = p.quantity;
    this.description = p.description;
    this.productId = p.productId;
    this.product = new Product(p.product);
  }
}

File: Production Back/Production.Domen/PlanItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Production.Domen
{
    public class PlanItem
    {
        public int Id { get; set; }
        public int AnnualProductionPlanId { get; set; }
        public double Quantity { get; set; }
        public string Description { get; set; }
        public int ProductId { get; set; }
        public Product Product { get; set; }
        public AnnualProductionPlan AnnualProductionPlan { get; set; }

    }


}
